lighthouse-core/gather/driver.js

```diff
@@ -550,6 +550,17 @@ class Driver {
     return {promise: Promise.resolve(), cancel() {}};
   }
 
+  /**
+   * Returns a promise that never resolves.
+   * Used for placeholder conditions that we don't want race with, but still want to satisfy the same
+   * interface.
+   * @return {{promise: Promise<void>, cancel: function(): void}}
+   */
+  _waitForever() {
+    return {promise: new Promise(() => null), cancel() {}};
```

Comment:
we probably still want to be able to cancel it though to avoid hanging async junk, just meant that the promise will "wait forever" until cancelled :)

sorry for the confusion!

Reply:
no problem, updated the PR

Reply:
Not sure yet why, but for some reason it seems like this also fixed some `NO_FCP` errors I was seeing in some pass (that I was still looking into)